gordo/machine/dataset/base.py

```diff
@@ -230,4 +231,9 @@ def _resample(
             resampled.columns = pd.MultiIndex.from_product(
                 [[series.name], resampled.columns], names=["tag", "aggregation_method"]
             )
-        return resampled.fillna(method="ffill")
+
+        # ensure interpolation max eight hours (8 x 60 minutes) forward
+        # given granularity in minutes
+        granularity = int(re.findall(r"\d+", resolution)[0])
```

Comment:
A resolution could have various numbers of [units](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Timedelta.html). My suggestion is:
```
         td = pd.Timedelta(resolution)
         granularity = td.total_seconds() / 60
```